### PR TITLE
fix(toast): switches toast[open] to use visibility hidden to fix overlay handling

### PR DESCRIPTION
**Pull request by benjamind**
## Description

Switches `sp-toast` to use the same behavior as `sp-tooltip` for controlling visibility with the `[open]` attribute. This switches from `display: none` to `visibility:hidden` which means the element now has size when hidden allowing the overlay behavior to properly measure it and therefore position it.

## Related issue(s)

#3510 

-

## Motivation and context

Fixes the placement in overlays.

## How has this been tested?

Added storybook to cover this case, and tested locally in Chrome.

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/toast/stories/toast.stories.ts [hunk 1]
```diff
@@ -104,3 +108,74 @@ export const Negative = (args: Properties): TemplateResult =>
...
+        ${(
+            [
+                ['bottom', ''],
+                ['left', 'negative'],
+                ['right', 'positive'],
+                ['top', 'info'],
+            ] as [Placement, string][]
+        ).map(([placement, variant]) => {
+            return html`
+                <overlay-trigger
```
Comment (Westbrook) [inline packages/toast/stories/toast.stories.ts:159]: We probably should not insinuate that Toast elements can/should be opened on `hover`. I think if this was set up at `type="click"` and the slot was `slot="click-content"` that we'd be pretty good to go here!

Comment (benjamind) [reply packages/toast/stories/toast.stories.ts:159]: haha yes good point.

Comment (benjamind) [reply packages/toast/stories/toast.stories.ts:159]: addressed in [592bdbd](https://github.com/adobe/spectrum-web-components/pull/3511/commits/592bdbdf7b59f9b27fade2d47a56e1f63dba3f6f)

**Review by Westbrook — COMMENTED** [review]
Looks a little _unique_, but otherwise looks pretty good to go!

If you could [cycle the VRTs](https://github.com/adobe/spectrum-web-components#keeping-ci-assets-updated), we'll be able to merge this and have it out to you sometime next week.

**Review by Westbrook — APPROVED** [review]
LGTM! Thanks @benjamind :shipit:

**Pull request merged**